__tests__/integration/atomic-swap.test.js

```diff
@@ -0,0 +1,477 @@
+
+    // Token 2
+    expect(finalTK2Balance1.available - startTK2Balance1.available).toEqual(15);
+    expect(finalTK2Balance2.available - startTK2Balance2.available).toEqual(-15);
+  });
+
+  it('should allow multisig wallets to participate', async () => {
+    // Get the balance state before the swap
+    // balances for wallet1 (HTR, token 1 and token 2)
+    const startHTRBalance1 = await wallet1.getBalance();
+    const startTK1Balance1 = await wallet1.getBalance(tokenTx1.hash);
+    const startTK2Balance1 = await wallet1.getBalance(tokenTx2.hash);
+
+    // balances for wallet2 (HTR, token 1 and token 2)
+    const startHTRBalance2 = await wallet2.getBalance();
+    const startTK1Balance2 = await wallet2.getBalance(tokenTx1.hash);
+    const startTK2Balance2 = await wallet2.getBalance(tokenTx2.hash);
+
+    // balances for walletMultisig (HTR, token 1 and token 2)
+    const startHTRBalanceMs = await walletMultisig.getBalance();
+    const startTK1BalanceMs = await walletMultisig.getBalance(tokenTx1.hash);
+    const startTK2BalanceMs = await walletMultisig.getBalance(tokenTx2.hash);
+
+    // walletMultisig proposes the transaction
+    // walletMultisig will:
+    //  - send 200 HTR
+    //  - receive 50 TKW1
+    //  - receive 50 TKW2
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        send_tokens: [{ value: 200 }], // HTR
+        receive_tokens: [
+          { token: tokenTx2.hash, value: 50 },
+          { token: tokenTx1.hash, value: 50 },
+        ],
+      })
+      .set({ 'x-wallet-id': walletMultisig.walletId });
+    loggers.test.insertLineToLog('atomic-swap[P2SH]: proposal', { body: response.body });
+    expect(response.body.isComplete).toBe(false);
+
+    let { data } = response.body;
+
+    // wallet1 updates the proposal
+    // wallet1 will:
+    //  - receive 60 HTR
+    //  - send 50 TKW1
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        partial_tx: data,
+        receive_tokens: [{ value: 60 }],
+        send_tokens: [{ token: tokenTx1.hash, value: 50 }],
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+    loggers.test.insertLineToLog('atomic-swap[P2SH]: update proposal 1', { body: response.body });
+    expect(response.body.isComplete).toBe(false);
+
+    data = response.body.data;
+
+    // wallet2 updates the proposal
+    // wallet2 will:
+    //  - receive 140 HTR
+    //  - send 50 TKW1
+    response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({
+        partial_tx: data,
+        receive_tokens: [{ value: 140 }],
+        send_tokens: [{ token: tokenTx2.hash, value: 50 }],
+      })
+      .set({ 'x-wallet-id': wallet2.walletId });
+    loggers.test.insertLineToLog('atomic-swap[P2SH]: update proposal 2', { body: response.body });
+    expect(response.body.isComplete).toBe(true);
+
+    data = response.body.data;
```

Comment:
We should use this data in the decode API to validate that everything is correct, especially all the change outputs. 

If there aren't change outputs, we should change the test to have changes for all wallets and validate each of the change outputs if their addresses belong to their own wallets, so we don't get to any bug similar to the multisig one.

Reply:
Done.

Reply:
Where was this done?

Reply:
Sorry, this was done for the multiple tokens tests and not for the multisig one.
Added this check for the multisig test.